src/matrix/matrixfdist.jl

```diff
@@ -0,0 +1,142 @@
+|\\mathbf{B}|^{n_2/2}|\\boldsymbol{\\Sigma}|^{(n_1-p-1)/2}|\\mathbf{B}+\\boldsymbol{\\Sigma}|^{-(n_1+n_2)/2}.
+```
+
+If the joint distribution ``p(\\boldsymbol{\\Psi},\\boldsymbol{\\Sigma})=p(\\boldsymbol{\\Psi})p(\\boldsymbol{\\Sigma}|\\boldsymbol{\\Psi})``
+is given by
+
+```math
+\\begin{align*}
+\\boldsymbol{\\Psi}&\\sim W_p(n_1, \\mathbf{B})\\\\
+\\boldsymbol{\\Sigma}|\\boldsymbol{\\Psi}&\\sim IW_p(n_2, \\boldsymbol{\\Psi}),
+\\end{align*}
+```
+
+then the marginal distribution of ``\\boldsymbol{\\Sigma}`` is
+``MF_{p}(n_1/2,n_2/2,\\mathbf{B})``.
+"""
+struct MatrixFDist{T <: Real, TW <: Wishart} <: ContinuousMatrixDistribution
+    W::TW
+    n2::T
+    logc0::T
+end
+
+#  -----------------------------------------------------------------------------
+#  Constructors
+#  -----------------------------------------------------------------------------
+
+function MatrixFDist(n1::Real, n2::Real, B::AbstractPDMat)
+
+    p = dim(B)
+
+    n1 > p - 1 || throw(ArgumentError("first degrees of freedom must be larger than $(p - 1)"))
+    n2 > p - 1 || throw(ArgumentError("second degrees of freedom must be larger than $(p - 1)"))
+
+    logc0 = matrixfdist_logc0(n1, n2, B)
+    T = Base.promote_eltype(n1, n2, logc0, B)
+    prom_B = convert(AbstractArray{T}, B)
+    W = Wishart(T(n1), prom_B)
+
+    MatrixFDist{T, typeof(W)}(W, T(n2), T(logc0))
+
+end
+
+MatrixFDist(n1::Real, n2::Real, B::Union{AbstractMatrix, LinearAlgebra.Cholesky}) = MatrixFDist(n1, n2, PDMat(B))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+ show(io::IO, d::MatrixFDist) = show_multline(io, d, [(:n1, d.W.df), (:n2, d.n2), (:B, d.W.S.mat)])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixFDist{T}}, d::MatrixFDist) where T <: Real
+    W = convert(Wishart{T}, d.W)
+    MatrixFDist{T, typeof(W)}(W, T(d.n2), T(d.logc0))
+end
+
+function convert(::Type{MatrixFDist{T}}, W::Wishart, n2, logc0) where T <: Real
+    WW = convert(Wishart{T}, W)
+    MatrixFDist{T, typeof(WW)}(WW, T(n2), T(logc0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+dim(d::MatrixFDist) = dim(d.W)
+
+size(d::MatrixFDist) = size(d.W)
+
+rank(d::MatrixFDist) = dim(d)
+
+insupport(d::MatrixFDist, Σ::AbstractMatrix) = isreal(Σ) && size(Σ) == size(d) && isposdef(Σ)
+
+params(d::MatrixFDist) = (d.W.df, d.n2, d.W.S)
+
+function mean(d::MatrixFDist)
+    p = dim(d)
+    n1, n2, B = params(d)
+
+    (n2 <= p + 1) ? throw(ArgumentError("mean only defined for df2 > dim + 1")) : (return (n1 / (n2 - p - 1)) * B.mat)
```

Comment:
could we maybe put this on several lines?
```suggestion
    if n2 <= p + 1
        throw(ArgumentError("mean only defined for df2 > dim + 1"))
    end
    return (n1 / (n2 - p - 1)) * B.mat
```